packages/core/core-v2/src/Parcel.js

```diff
@@ -112,72 +108,74 @@ export default class Parcel {
     }
   }
 
-  async updateGraph() {
-    // Process all changes that may update the middle of the the graph
-    this.updateQueue.start();
-    await this.updateQueue.onIdle();
-    this.updateQueue.pause();
+  async updateGraph({signal}: BuildOpts) {
+    for (let [id, node] of this.graph.invalidNodes) {
+      this.processNode(node, {signal, shallow: true});
+    }
+
+    await this.queue.run();
   }
 
-  async completeGraph({ signal }: BuildOpts) {
-    // AssetGraph keeps track of nodes that represent files that have not been transformed
-    // and dependencies that have not been resolved so that rebuilds start from the leaves
-    // of the graph
+  async completeGraph({signal}: BuildOpts) {
     for (let [id, node] of this.graph.incompleteNodes) {
-      this.mainQueue.add(() => this.processNode(node, { signal }));
+      this.processNode(node, {signal});
     }
 
-    // Process and build out the edges of the graph
-    this.mainQueue.start();
-    await this.mainQueue.onIdle();
-    this.mainQueue.pause();
-
-    if (signal.aborted) throw abortError;
+    await this.queue.run();
   }
 
-  processNode(node: Node, { signal }: BuildOpts) {
+  processNode(node: Node, buildOpts: BuildOpts) {
     switch (node.type) {
-      case 'dependency': return this.resolve(node.value, { signal });
-      case 'file': return this.transform(node.value, { signal });
-      default: throw new Error('Invalid Graph');
+      case 'dependency':
+        return this.resolve(node.value, buildOpts);
+      case 'file':
+        return this.transform(node.value, buildOpts);
+      default:
+        throw new Error('Invalid Graph');
     }
   }
 
-  async resolve(dep: Dependency, { signal }: BuildOpts) {
-    // console.log('resolving dependency', dep);
-    let resolvedPath = await this.resolverRunner.resolve(dep);
+  resolve(dep: Dependency, {signal}: BuildOpts) {
+    return this.queue.add(async () => {
+      // console.log('resolving dependency', dep);
+      let resolvedPath = await this.resolverRunner.resolve(dep);
+
+      if (signal.aborted) throw abortError;
 
-    let file = { filePath: resolvedPath };
-    if (signal && !signal.aborted) {
+      let file = {filePath: resolvedPath};
       let {newFile} = this.graph.updateDependency(dep, file);
 
       if (newFile) {
-        this.mainQueue.add(() => this.transform(newFile, { signal }));
+        this.transform(newFile, {signal});
         if (this.watcher) this.watcher.watch(newFile.filePath);
       }
-    }
+    });
   }
 
-  async transform(file: File, { signal, shallow }: BuildOpts) {
-    // console.log('transforming file', file);
-    let { children: childAssets } = await this.transformerRunner.transform(file);
-    if (signal && !signal.aborted) {
-      let { prunedFiles, newDeps } = this.graph.updateFile(file, childAssets);
+  transform(file: File, {signal, shallow}: BuildOpts) {
+    return this.queue.add(async () => {
+      // console.log('transforming file', file);
```

Comment:
remove console

Reply:
Can do, I was mostly keeping these logs around because I use them frequently for debugging. What are your thoughts on having some debugging logs?